### PR TITLE
Migrate iCub FT to have one NWS for each part

### LINKED ISSUE #485: Replace `analogServer` with `multipleanalogsensorsserver` for publishing FT data – Stint 2
This is the followup of #470 

We would like to:

> One thing that we may took the occasion to do know that we do a breaking change is that we could group in a single wrapper all the FT sensors of the leg, for consistency with what we do for other wrappers, such as controlboard.

### PR DESCRIPTION
**Pull request by Nicogene**
This is the follow-up of #505.

It fixes #485 

TO BE TESTED ON `iCubGenova02`

### CODE REVIEW

**Comment by Nicogene** [comment]
The errors reduced, but they are still there

immagine

File: iCubBarcelona01/icub_all.xml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="UTF-8" ?>
-<robot name="iCubBarcelona01" xmlns:xi="http://www.w3.org/2001/XInclude">
+<!DOCTYPE robot PUBLIC "-//YARP//DTD yarprobotinterface 3.0//EN" "http://www.yarp.it/DTD/yarprobotinterfaceV3.0.dtd">
+<robot name="iCubAnkara01" build="1" portprefix="icub" xmlns:xi="http://www.w3.org/2001/XInclude">
```
Comment (traversaro) [inline iCubBarcelona01/icub_all.xml:3]: I guess this is a spurious change?

**Comment by Nicogene** [comment]
The device `canBusAnalogSensor` had the same problem of `embObjStrain` (see #523), i.e. it does not implement the MAS interfaces.

For this reason [`f230652` (#517)](https://github.com/robotology/robots-configuration/pull/517/commits/f230652d6439f0764c052c6a50289761baf5a169) has replaced `canBusAnalogSensor` w/ `canBusFTSensor`(https://github.com/robotology/icub-main/pull/813) for the FT.
Moreover, now w/ this configuration we have the frame exposed also for the can robots, then in theory usable w/ ROS2 pipeline.

cc @traversaro @pattacini

**Comment by Nicogene** [comment]
Today I tested it on `iCubGenova02` rebasing this branch on https://github.com/robotology/robots-configuration/pull/523  and using this version of `icub-main` -> https://github.com/robotology/icub-main/issues/873

It worked flawlessly, after this fix: [`15003d5` (#517)](https://github.com/robotology/robots-configuration/pull/517/commits/15003d58d6305dc045dea8c778346da842332504) now the FT published divided per parts.


immagine


cc @traversaro @pattacini

**Comment by pattacini** [comment]
>TO BE TESTED ON iCubGenova11

Do we need to perform this test as well?

**Comment by pattacini** [comment]
CI just failed.

**Comment by Nicogene** [comment]
> > TO BE TESTED ON iCubGenova11
> 
> Do we need to perform this test as well?

`iCubGenova02` has both strain and strain2 and moreover has the remapping in the leg, the test should be sufficient.

> CI just failed.

I think I need to rebase it first

**Comment by pattacini** [comment]
Still failing.

Some stray logs from `iCubErzelli01`:

```
[ERROR] |yarp.device.multipleanalogsensorsremapper| Impossible to find sensor name r_leg_ft_sensor , exiting.
[ERROR] |yarp.device.multipleanalogsensorsremapper|     Names of available sensors are:
[ERROR] Device right_leg-FT_remapper cannot execute attach
[ERROR] Cannot run attach action on device right_leg-FT_remapper
[INFO] Executing attach action, level 5 on device left_leg-FT_remapper with parameters [("networks" = "(left_leg-eb6-j0_3-strain left_leg-eb7-j4_5-strain)"), ("left_leg-eb6-j0_3-strain" = "left_leg-eb6-j0_3-strain"), ("left_leg-eb7-j4_5-strain" = "left_leg-eb7-j4_5-strain")]
[ERROR] |yarp.device.multipleanalogsensorsremapper| Impossible to find sensor name l_leg_ft_sensor , exiting.
[ERROR] |yarp.device.multipleanalogsensorsremapper|     Names of available sensors are:
[ERROR] Device left_leg-FT_remapper cannot execute attach
[ERROR] Cannot run attach action on device left_leg-FT_remapper
```

**Comment by traversaro** [comment]
For iCubErzelli01, I guess we need to align the id of the ft sensors with the name used in the URDF, see https://github.com/robotology/robots-configuration/commit/cdf0cc6d0b06ca5f461f4d48bff689953541f601 .

**Comment by Nicogene** [comment]
The failure shoudl be fixed by [`4f8037c` (#517)](https://github.com/robotology/robots-configuration/pull/517/commits/4f8037c6fc1634e84833fb63fb4ee528a26149a1). The problem was the remapping of the temperature measure for the sensor that does not have it (e.g. strain)

**Review by pattacini — APPROVED** [review]

**Pull request merged**